src/Disks/ObjectStorages/DiskObjectStorage.cpp

```diff
@@ -528,6 +536,11 @@ std::unique_ptr<ReadBufferFromFileBase> DiskObjectStorage::readFile(
         read_hint,
         file_size);
 }
+catch (S3Exception & e)
+{
+    e.addMessage("on file read operation in DiskObjectStorage");
+    throw;
+}
```

Comment:
Why is it allowed? I mean the syntax

```
void f()
try
{
    ...
}
catch
...
```

instead of

```
void f()
{
    try
    {
        ...
    }
    catch
    ...
}
```

Reply:
It's called [function-try-block](https://en.cppreference.com/w/cpp/language/function-try-block) and it mostly should be used with constructors, and it seems for regular functions that doesn't make much sense.